app/src/test/java/aero/minova/cas/app/extension/UserGroupExtensionTest.java

```diff
@@ -0,0 +1,81 @@
+import org.springframework.test.context.ActiveProfiles;
+
+import com.google.gson.Gson;
+
+import aero.minova.cas.CoreApplicationSystemApplication;
+import aero.minova.cas.api.domain.Table;
+import aero.minova.cas.api.domain.Value;
+import aero.minova.cas.api.domain.ValueDeserializer;
+import aero.minova.cas.api.domain.ValueSerializer;
+import aero.minova.cas.service.repository.UserGroupRepository;
+
+@SpringBootTest(classes = CoreApplicationSystemApplication.class)
+@ActiveProfiles("test")
+@DirtiesContext(classMode = ClassMode.BEFORE_EACH_TEST_METHOD)
+class UserGroupExtensionTest {
+
+	String insertTable = "{\n" + "            \"name\": \"xpcasInsertUserGroup\",\n" + "            \"columns\": [\n" + "                {\n"
+			+ "                    \"name\": \"KeyLong\",\n" + "                    \"type\": \"INTEGER\",\n"
+			+ "                    \"outputType\": \"OUTPUT\"\n" + "                },\n" + "                {\n"
+			+ "                    \"name\": \"KeyText\",\n" + "                    \"type\": \"STRING\"\n" + "                },\n" + "                {\n"
+			+ "                    \"name\": \"Description\",\n" + "                    \"type\": \"STRING\"\n" + "                },\n" + "                {\n"
+			+ "                    \"name\": \"UserCode\",\n" + "                    \"type\": \"STRING\"\n" + "                },\n" + "                {\n"
+			+ "                    \"name\": \"SecurityToken\",\n" + "                    \"type\": \"STRING\"\n" + "                }\n" + "            ],\n"
+			+ "            \"rows\": [\n" + "                {\n" + "                    \"values\": [\n" + "                        null,\n"
+			+ "                        \"s-Group\",\n" + "                        null,\n" + "                        null,\n"
+			+ "                        \"s-Token\"\n" + "                    ]\n" + "                }\n" + "            ]\n" + "        }";
+
+	String updateTable = "{\"name\":\"xpcasUpdateUserGroup\",\"columns\":[{\"name\":\"KeyLong\",\"type\":\"INTEGER\"},{\"name\":\"KeyText\",\"type\":\"STRING\"},{\"name\":\"Description\",\"type\":\"STRING\"},{\"name\":\"UserCode\",\"type\":\"STRING\"},{\"name\":\"SecurityToken\",\"type\":\"STRING\"}],\"rows\":[{\"values\":[\"n-2\",\"s-Group\",\"s-Des\",null,\"s-Token\"]}]}";
+
+	String readTable = "{\"name\":\"xpcasReadUserGroup\",\"columns\":[{\"name\":\"KeyLong\",\"type\":\"INTEGER\",\"outputType\":\"OUTPUT\"},{\"name\":\"KeyText\",\"type\":\"STRING\",\"outputType\":\"OUTPUT\"},{\"name\":\"Description\",\"type\":\"STRING\",\"outputType\":\"OUTPUT\"},{\"name\":\"UserCode\",\"type\":\"STRING\",\"outputType\":\"OUTPUT\"},{\"name\":\"SecurityToken\",\"type\":\"STRING\",\"outputType\":\"OUTPUT\"}],\"rows\":[{\"values\":[\"n-2\",null,null,null,null]}]}";
+
+	String deleteTable = "{\"name\":\"xpcasDeleteUserGroup\",\"columns\":[{\"name\":\"KeyLong\",\"type\":\"INTEGER\"}],\"rows\":[{\"values\":[\"n-2\"]}]}";
```

Comment:
Diese Strings können wir als Dateien in src/test/resources legen und auslesen lassen.
Dafür kann die readTableFromExampleJson()-Methode  aus dem BaseTest kann dafür verwendet werden. Siehe Klasse ErrorMessageTest.java.